solutions/CSES Pizzeria Queries.mdx

```diff
@@ -0,0 +1,82 @@
+---
+id: cses-2206
+source: CSES
+title: Pizzeria Queries
+author: Dong Liu
+---
+
+# Solution
+
+When we query the minimum cost to buy a pizza at point $i$, we can split it into two cases ($i$$\rightarrow$$j$)
+1. going downwards ($j$<$i$) : $cost$ would be $p_j-j+i$
+2. going upwards ($j$>$i$) : $cost$ would be $p_j+j-i$
+
+Notice that we can caculate the $+i$ or $-i$ after the query since it is constant.
+
+Thus, if we keep two min segtrees (one for going downwards and one for going upwards), we can keep track of the minumum cost by setting each value in the downwards segtree to $p_j-j$ and each value in the upwards segtree to $p_j+j$.
+
+To query, we can query the minimum value in the
+1. downwards segtree: $min(1,\dots,i)$ + i
+2. upwards segtree: $min(i,\dots,N)$ - i
```

Comment:
```suggestion
1. downwards segtree: $\min(1,\dots,i) + i$
2. upwards segtree: $\min(i,\dots,N) - i$
```
I think that just saying $\min(1, \dots, i)$ makes it sound a bit like you're querying the minimum value out of the list of numbers from 1 to i